### PR TITLE
TDEVOPS-1466 | Vulnerability management

### PR DESCRIPTION
**Pull request by atharvatessell**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

Dependent Services #
- [ ] Governance
- [ ] IAM
- [ ] Deployer
- [ ] DB Systems
- [ ] Tenant
- [ ] Software Library
- [ ] Backup
- [ ] Conductor
- [ ] Monitoring
- [ ] DMM

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Environment**:

- [ ] Dev
- [ ] Staging

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/master_build_pr.yml [hunk 1]
```diff
@@ -12,7 +12,14 @@ jobs:
 
   build-amd:
     if: ${{inputs.type != 'arm' }}
-    runs-on: self-hosted
+    runs-on: self-hosted-test
+    env:
+      API_KEY: ${{secrets.CONVOY_AUTH_TOKEN}}
+      CHANNEL_ID: ${{vars.SKGUPTA_CHANNEL_SLACK_ID}}
+      DEVOPS_TEAM: ${{vars.DEVOPS_TEAM_SLACK_ID}}
+      IMAGE_SCAN_API_URL: http://${{vars.CONVOY_DEV_API_ENDPOINT}}/devops/code-scan/vulnerabilities/validate
```
Comment (varenyatessell) [inline .github/workflows/master_build_pr.yml:20]: Change `SLACK_CHANNEL_ID` `SLACK_TOKEN` and `CONVOY_DEV_API_ENDPOINT`

File: .github/workflows/master_pr.yml [hunk 2]
```diff
@@ -71,7 +71,7 @@ jobs:
 
   build:
     needs: build-file
-    uses: TessellDevelopment/convoy-ci/.github/workflows/master_build_pr.yml@main
+    uses: TessellDevelopment/convoy-ci/.github/workflows/master_build_pr.yml@TDEVOPS-1466
```
Comment (varenyatessell) [inline .github/workflows/master_pr.yml:74]: Change ref to main

File: .github/workflows/master_build_pr.yml [hunk 3]
```diff
@@ -35,6 +42,70 @@ jobs:
...
+              print(response.text)
+            elif response.status_code == 409:
+              print(response.json())
+              print("Get approval for these vulnerabilities or fix them")
+              exit(1)
+            else:
+              print(response.status_code)
+              print(response.text)
+              post_to_slack(response.text)
+              exit(1)                      #check to fail or not
```
Comment (atharvatessell) [inline .github/workflows/master_build_pr.yml:103]: check to fail or not

File: .github/workflows/master_build_pr.yml [hunk 4]
```diff
@@ -35,6 +42,70 @@ jobs:
...
+          except Exception as e:
+            print(e)
+            post_to_slack(e)
+            exit(1)               #check to fail or not
```
Comment (atharvatessell) [inline .github/workflows/master_build_pr.yml:107]: check to fail or not

File: .github/workflows/master_build_pr.yml (hunk 1 above)
Comment (atharvatessell) [reply .github/workflows/master_build_pr.yml:20]: api dev->prod,
slack-channel skgupta->convoy-alerts
token   ideally should be working with current one

File: .github/workflows/master_build_pr.yml [hunk 5]
```diff
@@ -13,6 +13,13 @@ jobs:
   build-amd:
     if: ${{inputs.type != 'arm' }}
     runs-on: self-hosted
+    env:
+      API_KEY: ${{secrets.CONVOY_AUTH_TOKEN}}
+      CHANNEL_ID: ${{vars.SKGUPTA_CHANNEL_SLACK_ID}}
```
Comment (sundeeptessell) [inline .github/workflows/master_build_pr.yml:18]: Use `vars.CONVOY_ALERTS_SLACK_ID`

File: .github/workflows/master_build_pr.yml [hunk 6]
```diff
@@ -35,6 +42,70 @@ jobs:
             echo "$FILE"
             echo "Starting Docker"
             docker build -f $FILE -t $IMAGE_NAME .
+            imageScan $IMAGE_NAME
+            set +e
+          }
+          imageScan(){
```
Comment (sundeeptessell) [inline .github/workflows/master_build_pr.yml:48]: rename it to `dockerImageScan` or `containerImageScan`. With packer, `image` will be ambiguous

File: .github/workflows/master_build_pr.yml [hunk 7]
```diff
@@ -13,6 +13,13 @@ jobs:
   build-amd:
     if: ${{inputs.type != 'arm' }}
     runs-on: self-hosted
+    env:
+      API_KEY: ${{secrets.CONVOY_AUTH_TOKEN}}
```
Comment (sundeeptessell) [inline .github/workflows/master_build_pr.yml:17]: Rename to CONVOY_API_KEY

File: .github/workflows/master_build_pr.yml [hunk 8]
```diff
@@ -35,6 +42,70 @@ jobs:
...
             docker build -f $FILE -t $IMAGE_NAME .
+            imageScan $IMAGE_NAME
+            set +e
+          }
+          imageScan(){
+            set -e
+            IMAGE_NAME="$1"
+            trivy image -f json -o trivy_image_scan_result.json  --severity HIGH,CRITICAL --scanners vuln $IMAGE_NAME:latest
+            dockle -f json -o dockle_image_scan_result.json $IMAGE_NAME:latest
+            python3 - <<EOF
```
Comment (sundeeptessell) [inline .github/workflows/master_build_pr.yml:53]: You are  Passed - You can define a python method  inside a anonymous python block -> which is inside a bash function call - which inturn is inside a bash script -> which is called via the nodejs (a.k.a Github Actions).

Now split this so humans can understand it.

**Review by sundeeptessell — APPROVED** [review]

**Pull request merged**